packages/@aws-cdk/aws-scheduler-targets-alpha/lib/sns-publish.ts

```diff
@@ -1,37 +1,20 @@
-import { ISchedule, IScheduleTarget } from '@aws-cdk/aws-scheduler-alpha';
-import { Names } from 'aws-cdk-lib';
+import { IScheduleTarget } from '@aws-cdk/aws-scheduler-alpha';
 import { IRole } from 'aws-cdk-lib/aws-iam';
 import * as sns from 'aws-cdk-lib/aws-sns';
 import { ScheduleTargetBase, ScheduleTargetBaseProps } from './target';
-import { sameEnvDimension } from './util';
 
 /**
  * Use an Amazon SNS topic as a target for AWS EventBridge Scheduler.
  */
 export class SnsPublish extends ScheduleTargetBase implements IScheduleTarget {
   constructor(
     private readonly topic: sns.ITopic,
-    private readonly props: ScheduleTargetBaseProps = {},
+    props: ScheduleTargetBaseProps = {},
```

Comment:
> curious why private readonly is removed here?

Since we removed usage of `props` (the validation clauses), TypeScript was complaining "Property 'props' is declared but its value is never read."  If needed in the future we could declare it as a class variable again.

> Also some props are assigned to {} while others are not (e.g kinesis-stream-put-record). Just curious what's the difference here?
Oh good catch, I don't think there's a difference and I meant to assign `{}` to props by default for all targets so user doesn't need to pass in empty object. Will update!

Reply:
Oh actually it looks like most targets use `ScheduleTargetBaseProps` but some target classes like Kinesis stream have additional props. In the case of `kinesis-stream-put-record`, the additional prop (`partitionKey`) is required so we cannot pass in empty object as the default.

Reply:
Updated!